shell/platform/android/test/io/flutter/embedding/engine/systemchannels/SettingsChannelTest.java

```diff
@@ -0,0 +1,70 @@
+package io.flutter.embedding.engine.systemchannels;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Mockito.eq;
+import static org.mockito.Mockito.isNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import android.annotation.TargetApi;
+import android.util.DisplayMetrics;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.plugin.common.BasicMessageChannel;
+import java.nio.ByteBuffer;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.robolectric.annotation.Config;
+
+@Config(manifest = Config.NONE)
+@RunWith(AndroidJUnit4.class)
+public class SettingsChannelTest {
+  @Test
+  @TargetApi(33)
+  @Config(sdk = 33)
+  public void setDisplayMetricsDoesNothingOnAPILevel33() {
+    final DartExecutor executor = mock(DartExecutor.class);
+    executor.onAttachedToJNI();
+    final SettingsChannel settingsChannel = new SettingsChannel(executor);
+
+    final ArgumentCaptor<ByteBuffer> messageCaptor = ArgumentCaptor.forClass(ByteBuffer.class);
+
+    settingsChannel.startMessage().setDisplayMetrics(mock(DisplayMetrics.class)).send();
+
+    verify(executor).send(eq("flutter/settings"), messageCaptor.capture(), isNull());
+  }
+
+  @Test
```

Comment:
Should this test have a config = 34?

Reply:
I'll give that a try. But iirc Robolectric does not support that api level yet and the test will fail.

Reply:
it gave me 
```
io.flutter.embedding.engine.systemchannels.SettingsChannelTest > configurationQueueWorks FAILED
    java.lang.IllegalArgumentException: API level 34 is not available
        at org.robolectric.plugins.UnknownSdk.verifySupportedSdk(UnknownSdk.java:45)
        at org.robolectric.RobolectricTestRunner.getSandbox(RobolectricTestRunner.java:298)
        at org.robolectric.RobolectricTestRunner.getSandbox(RobolectricTestRunner.java:66)
        at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:244)
```

Reply:
But you are calling an api 34 method. If roboletric does not support api 34 then update the dependency. Or at minimum use config minSdk to api 33 so that when api 34 is added this test starts running against api 34.

Reply:
None of these methods directly or indirectly called is API 34 only I think.

Reply:
If this code does not indirectly call 34 then shouldnt there be a test for the opposite of setDisplayMetricsDoesNothingOnAPILevel33. Something like setDisplayMetricsDoesXyzBehaviorOnApiLevel34?

Reply:
I am sorry I am doing a poor job of communicating. Yes from your comments in this pr I see that the existing robolectric dependency does not support api 34. I believe you are adding code that splits and takes a different branch on api 34. If that is the case then you should add a test to cover this new feature otherwise it will be untested. If you need to update a dependency to test the new code that is what should happen. Otherwise this is new code with an untested branch which we want to avoid.

Reply:
edit: I was mistaken about roboletric api 34 support. 

OK, can you file a follow up bug to add a test when we have a roboletric version that supports api 34?